lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -831,32 +624,75 @@ unsigned SliceEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
   return product<unsigned>(getElemsPerThread(shape, eltTy));
 }
 SmallVector<unsigned> SliceEncodingAttr::getCTASplitNum() const {
-  return ::getCTASplitNum(*this);
+  return ::getCTASplitNum(getParent());
 }
```

Comment:
that looks different than what was implemented:
```
 else if (auto sliceLayout = layout.dyn_cast<SliceEncodingAttr>()) {
    res = getCTASplitNum(sliceLayout.getParent());
    res.erase(res.begin() + sliceLayout.getDim());
  }
```

Reply:
Yes, this is the root cause of the issue. I fixed this in new PR.